### PR TITLE
keysapce: wait region split when creating keyspace

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref #6231

It is a pick from pd-cse.

There are some conflict in pkg/keyspace/keyspace.go, which we need to review carefully

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/keyspace/keyspace.go [hunk 1]
```diff
@@ -252,27 +267,81 @@ func (manager *Manager) saveNewKeyspace(keyspace *keyspacepb.KeyspaceMeta) error
...
+	}
+	err = cl.GetRegionLabeler().SetLabelRule(keyspaceRule)
+	if err != nil {
+		log.Warn("[keyspace] failed to add region label for keyspace",
 			zap.Uint32("keyspaceID", id),
-			zap.Any("LabelRule", keyspaceRule),
+			zap.Error(err),
 		)
-		return nil
+		return err
```
Comment (lhy1024) [inline pkg/keyspace/keyspace.go:287]: when meeting this error it will return nil in pd master, but it returns an error in pd-cse

**Review by zeminzhou — APPROVED** [review]
LGTM

**Review by zeminzhou — APPROVED** [review]
LGTM

**Review by binshi-bing — APPROVED** [review]

File: pkg/keyspace/keyspace.go [hunk 2]
```diff
@@ -252,27 +267,81 @@ func (manager *Manager) saveNewKeyspace(keyspace *keyspacepb.KeyspaceMeta) error
...
 	}
-	return errors.New("cluster does not support region label")
+	defer func() {
+		if err != nil {
+			cl.GetRegionLabeler().DeleteLabelRule(keyspaceRule.ID)
+		}
+	}()
+
+	if waitRegionSplit {
+		ranges := keyspaceRule.Data.([]*labeler.KeyRangeRule)
```
Comment (rleungx) [inline pkg/keyspace/keyspace.go:296]: Should we check the length?

File: pkg/keyspace/keyspace.go [hunk 3]
```diff
@@ -252,27 +267,81 @@ func (manager *Manager) saveNewKeyspace(keyspace *keyspacepb.KeyspaceMeta) error
...
+				return
+			}
+			log.Info("[keyspace] wait reigon split successfully", zap.Uint32("keyspaceID", id))
+			break
+		}
+	}
+
+	log.Info("[keyspace] added region label for keyspace",
+		zap.Uint32("keyspaceID", id),
+		zap.Any("LabelRule", keyspaceRule),
```
Comment (rleungx) [inline pkg/keyspace/keyspace.go:341]:
```suggestion
		zap.Uint32("keyspace-id", id),
		zap.Any("label-rule", keyspaceRule),
```

File: pkg/keyspace/keyspace.go (hunk 2 above)
Comment (lhy1024) [reply pkg/keyspace/keyspace.go:296]: I added. cc @zeminzhou

**Review by rleungx — APPROVED** [review]

**Pull request merged**